lib/index.js

```diff
@@ -210,16 +246,48 @@ export function configureUnifiedLanguageServer(
     }
   }
 
-  connection.onInitialize(() => ({
-    capabilities: {
-      textDocumentSync: TextDocumentSyncKind.Full,
-      documentFormattingProvider: true,
-      codeActionProvider: {
-        codeActionKinds: [CodeActionKind.QuickFix],
-        resolveProvider: true
+  connection.onInitialize((event) => {
+    if (event.workspaceFolders) {
+      workspaces = event.workspaceFolders.map((d) => d.uri)
+    }
+
+    if ((!workspaces || workspaces.length === 0) && event.rootUri) {
+      workspaces = [event.rootUri]
+    }
```

Comment:
Given the suggestion above, this would become:

```js
if (event.workspaceFolders && event.workspaceFolders.length) {
  addWorkspaces(event.workspaceFolders)
} else if (event.rootUri) {
  addWorkspaces([event.rootUri])
}
```

Reply:
Why use special functions when a loop can do it (especially because there are different values: the first is an array of objects, the second a string, so the first will still need a loop/map"?)